barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/mega_flavor.hpp

```diff
@@ -550,6 +550,9 @@ class MegaFlavor {
         VerificationKey(ProvingKey& proving_key)
         {
             set_metadata(proving_key);
+            if (proving_key.commitment_key == nullptr) {
```

Comment:
Will you please add a comment why this is needed i.e. why this could be a nullptr at this point